### PR TITLE
Group "utility" operations into same file

### PR DESCRIPTION
**Pull request by AnuravModak**
Towards  #2962 .

So, this is the draft for this issue.

Changes made:
grouping all three ops:

- Barrier
- WireCut
- Snapshot

into a new file  `pennylane/ops/meta.py`.

Also, moved tests from `tests/ops/qubit/test_non_parametric_opy.py` and `tests/ops/test_snapshot.py` to a new file `tests/ops/test_meta.py`.

Please let me know if i need to do anything else.

### CODE REVIEW

**Comment by AnuravModak** [comment]
Hey, @albi3ro , there is kind of deadlock or conflicting situation between pylint and other tests here. 
So, when i am including this `from .meta import Snapshot` at line 48 in `pennylane\ops\cv.py` , pylint issues are failing but when i am removing that line other tests are failing, kindly help me here.

**Comment by albi3ro** [comment]
> Hey, @albi3ro , there is kind of deadlock or conflicting situation between pylint and other tests here. So, when i am including this `from .meta import Snapshot` at line 48 in `pennylane\ops\cv.py` , pylint issues are failing but when i am removing that line other tests are failing, kindly help me here.

Thanks for opening this PR 🎉 

Regarding the snapshot import in `cv.py`, I'm fine with just silencing the pylint warning with `# pylint: disable=unused-import`. That error suppression existed before and it's not worth the effort to fix.

**Review by albi3ro — COMMENTED** [review]
Looking great.  Let's just add a changelog entry under "Breaking changes" and I'll be happy to approve.

This could potentially also be called an "Improvement", but I think its safer to call it a "breaking change".

**Review by albi3ro — APPROVED** [review]
Looks great! 🎉 Thanks for completing this issue :)

**Review by timmysilv — APPROVED** [review]
looks great!

**Comment by AnuravModak** [comment]
Hi @albi3ro and @timmysilv,

I've completed two good first issues in this repository and have become well-acquainted with most of the components. I'm now eager to take on some intermediate-level tasks. Please let me know if there are any available.

**Comment by albi3ro** [comment]
@AnuravModak  I just opened this issue: https://github.com/PennyLaneAI/pennylane/issues/4824 which might be a good candidate.

File: doc/releases/changelog-dev.md [hunk 1]
```diff
@@ -30,6 +30,10 @@
 * `qml.gradients.pulse_generator` has become `qml.gradients.pulse_odegen` to adhere to paper naming conventions.
   [(#4769)](https://github.com/PennyLaneAI/pennylane/pull/4769)
 
+* Non-parametric-ops such as `Barrier`, `Snapshot` and `Wirecut` have been grouped together and moved to `pennylane/ops/meta.py`.
+  Additionally, the relevant tests have been organized and placed in a new file, `tests/ops/test_meta.py` .
+  [(#4789)](https://github.com/PennyLaneAI/pennylane/pull/4789)
+
```
Comment (trbromley) [inline doc/releases/changelog-dev.md:36]: @albi3ro is this a breaking change, or could it just be in "improvements"?

Comment (albi3ro) [reply doc/releases/changelog-dev.md:36]: either.

Comment (trbromley) [reply doc/releases/changelog-dev.md:36]: Thanks!

**Pull request merged**